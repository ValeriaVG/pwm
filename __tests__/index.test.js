const PWM = require('../index')
describe('PWM', () => {
  it('creates and starts manager', done => {
    const n = 10
    let _done = 0
    let inputs = []
    let outputs = []
    const getBatch = async () => {
      if (_done >= 100) return []
      let a = []
      for (let i = 0; i < n; i++) {
        a.push({ a: Math.random(), b: Math.random() })
      }
      _done += n
      inputs = inputs.concat(a)
      return a
    }

    const pwm = new PWM({
      path: 'example/workerWithErrors.js',
      workers: 10,
      nextBatch: () => getBatch(10),
      done: ({ input, output }) => {
        if (input.b > 0.5) expect(output.error).toBe('Random error happened')
        else expect(output).toBe(input.a / input.b)
        outputs.push(output)
        if (outputs.length === inputs.length) done()
      },
    })
  })
  it('throws errors if not all properties filled', () => {
    expect(() => {
      const pwm = new PWM()
    }).toThrowErrorMatchingSnapshot()
  })
})
