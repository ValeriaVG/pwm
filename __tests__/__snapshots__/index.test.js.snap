// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PWM throws errors if not all properties filled 1`] = `"You must provide path, nextBatch and done"`;
